pds_doi_core/input/osti_input_util.py

```diff
@@ -0,0 +1,138 @@
+                if my_root.xpath('record/publication_date'):
+                    response_dict['publication_date'] = my_root.xpath('record/publication_date')[element_count].text
+                if my_root.xpath('record/product_type'):
+                    response_dict['product_type'] = my_root.xpath('record/product_type')[element_count].text
+                if my_root.xpath('record/product_type_specific'):
+                    response_dict['product_type_specific'] = my_root.xpath('record/product_type_specific')[element_count].text
+                if my_root.xpath('record/date_first_registered'):
+                    response_dict['date_first_registered'] = my_root.xpath('record/date_first_registered')[element_count].text
+                if my_root.xpath('record/date_last_registered'):
+                    response_dict['date_last_registered'] = my_root.xpath('record/date_last_registered')[element_count].text
+                if my_root.xpath('record/date_record_added'):
+                    response_dict['date_record_added'] = my_root.xpath('record/date_record_added')[element_count].text
+                if my_root.xpath('record/authors'):
+                    num_authors = len(my_root.xpath('record/authors')[0])
+                    response_dict['authors'] = []
+                    for ii in range(num_authors):
+                        one_author = {}
+                        one_author['first_name']  = (my_root.xpath('record/authors')[element_count]).xpath('author/first_name')[ii].text
+                        one_author['last_name']   = (my_root.xpath('record/authors')[element_count]).xpath('author/last_name') [ii].text
+                        response_dict['authors'].append(one_author)
+
+                if my_root.xpath('record/related_identifiers'):
+                    num_identifiers = len(my_root.xpath('record/related_identifiers')[0])
+                    response_dict['related_identifiers'] = []
+                    for ii in range(num_identifiers):
+                        one_identifier = {}
+                        one_identifier['identifier_type']  = (my_root.xpath('record/related_identifiers')[element_count]).xpath('related_identifier/identifier_type') [ii].text
+                        one_identifier['identifier_value'] = (my_root.xpath('record/related_identifiers')[element_count]).xpath('related_identifier/identifier_value')[ii].text
+                        one_identifier['relation_type']    = (my_root.xpath('record/related_identifiers')[element_count]).xpath('related_identifier/relation_type')[ii].text
+                        response_dict['related_identifiers'].append(one_identifier)
+
+                o_response_dicts.append(response_dict)  # App the dictionary response_dict to o_response_dicts list.
+                element_count += 1
+
+        return o_response_dicts
+
+    def process_pds4_fields(self, pds4_fields):
```

Comment:
Apparently the methods after are not used. If you confirm can you remove them ?

Reply:
They are being used.  Can confirm:

{pds-dev-el7.jpl.nasa.gov}/home/qchau/sandbox/pds-doi-service/pds_doi_core 111 % grep -n get_publication_date */osti_input_util.py
78:        doi_field_value_dict['publication_date'] = self.get_publication_date(pds4_fields)  # datetime object
93:    def get_publication_date(self, pds4_fields):

{pds-dev-el7.jpl.nasa.gov}/home/qchau/sandbox/pds-doi-service/pds_doi_core 112 % grep -n get_keywords */osti_input_util.py
89:        doi_field_value_dict['keywords'] = self.get_keywords(pds4_fields)
101:    def get_keywords(self, pds4_fields):

Reply:
Yes they are used in this file but process_pds4_fields from this object is not used in the project. That is what my IDE says.

Reply:
And now the method read_xml is also not used. It looks like finally the object is not used anymore. I remove it. The parser in the output package does what is needed

Reply:
I removed it in branch #24_46_tl